flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -337,6 +362,47 @@ public TableConfig getConfig() {
 		return tableConfig;
 	}
 
+	@Override
+	public JobExecutionResult execute(String jobName) throws Exception {
+		translate(bufferedModifyOperations);
+		return execEnv.execute(jobName);
```

Comment:
`bufferedModifyOperations` must be cleared here, otherwise the `bufferedModifyOperations` will be translated again and again if table env is used for multiple job. 
e.g.
val tableEnv = // shouldTranslateEagerly is false
tableEnv.sqlUpdate("insert into table1 select ...");
tableEnv.execute("job1")
tableEnv.sqlUpdate("insert into table2 select ..."); 
tableEnv.execute("job2")

Reply:
good catch!